src/api.js

```diff
@@ -789,7 +789,13 @@ API.prototype.declare = function(options, handler) {
   if (this._entries.filter(entry => entry.route == options.route && entry.method == options.method).length > 0) {
     throw new Error('Identical route and method declaration.');
   }
-  this._entries.push(options);
+  if (this._entries.map(a => a.name).some(function(element) {
+    return element == options.name;
+  })) {
+    throw new Error('This function has already been declared.');
+  } else {
+    this._entries.push(options);
+  }
```

Comment:
The true branch of this conditional doesn't return (it throws), so there's no need for an `else` here.